lib/pbench/server/filetree.py

```diff
@@ -237,30 +237,27 @@ def extract(self, path: str) -> str:
         except Exception as exc:
             raise MetadataError(self.tarball_path, exc)
 
-    def get_metadata(self) -> Dict[str, Any]:
+    def get_metadata(self) -> Optional[JSONOBJECT]:
```

Comment:
The `try` block in upload will catch a missing key, in case we can't extract metadata. The apparent willingness to set the metadata key to value `None` is misleading since obviously we'll never get that far if we can't excavate metadata.

You're probably right that it's better to let that outer `try` catch an exception during the process of acquiring metadata; I think I'll just remove this `try` block entirely, which does probably make the `Optional`, um, optional...